### ISSUE TITLE
Doesn't work with my Asus OpenVPN Router

### ISSUE THREAD

**Issue by NicolasLoew** [angristan/openvpn-install#162]
Any idea why it wouldn't be working with my Asus Open-VPN Router? Works just fine on desktop/phone with respective apps/programms but won't work at router level :/

**Comment by angristan** [comment]
Do you have any log?

**Comment by NicolasLoew** [comment]
There you go (I'm running the openvpn server on CentOS 6 64 bit):

Router Logs:
Feb 14 12:39:12 rc_service: httpd 212:notify_rc restart_vpncall
Feb 14 12:39:16 rc_service: httpd 212:notify_rc restart_vpncall
Feb 14 12:39:16 openvpn[520]: Options error: Unrecognized option or missing parameter(s) in config.ovpn:30: tls-version-min (2.3.2)
Feb 14 12:39:16 openvpn[520]: Use --help for more information.

**Comment by NicolasLoew** [comment]
I didn't touch the .ovpn file though so it is exactly as it is crafted by your script.

**Comment by NicolasLoew** [comment]
Seems like I need higher TLS version?

**Comment by angristan** [comment]
Remove the `tls-version-min` line from the configuration file. The OpenVPN client in your router is too old

**Comment by NicolasLoew** [comment]
Ok will try thanks! :D

**Comment by NicolasLoew** [comment]
Now it's even more TLS errors:

```
Feb 14 12:57:07 openvpn[810]: Socket Buffers: R=[116736->131072] S=[116736->131072]
Feb 14 12:57:07 openvpn[810]: UDPv4 link local: [undef]
Feb 14 12:57:07 openvpn[810]: UDPv4 link remote: [AF_INET]1xx.xx.247.13:1194
Feb 14 12:57:07 openvpn[810]: TLS_ERROR: BIO read tls_read_plaintext error: error:140830B5:SSL routines:ssl3_client_hello:no ciphers available
Feb 14 12:57:07 openvpn[810]: TLS Error: TLS object -> incoming plaintext read error
Feb 14 12:57:07 openvpn[810]: TLS Error: TLS handshake failed
Feb 14 12:57:07 openvpn[810]: SIGUSR1[soft,tls-error] received, process restarting
Feb 14 12:57:07 openvpn[810]: Restart pause, 2 second(s)
Feb 14 12:57:09 openvpn[810]: Socket Buffers: R=[116736->131072] S=[116736->131072]
Feb 14 12:57:09 openvpn[810]: UDPv4 link local: [undef]
Feb 14 12:57:09 openvpn[810]: UDPv4 link remote: [AF_INET]1xx.xx.247.13:1194
Feb 14 12:57:09 openvpn[810]: TLS_ERROR: BIO read tls_read_plaintext error: error:140830B5:SSL routines:ssl3_client_hello:no ciphers available
Feb 14 12:57:09 openvpn[810]: TLS Error: TLS object -> incoming plaintext read error
Feb 14 12:57:09 openvpn[810]: TLS Error: TLS handshake failed
Feb 14 12:57:09 openvpn[810]: SIGUSR1[soft,tls-error] received, process restarting
Feb 14 12:57:09 openvpn[810]: Restart pause, 2 second(s)
Feb 14 12:57:11 openvpn[810]: Socket Buffers: R=[116736->131072] S=[116736->131072]
Feb 14 12:57:11 openvpn[810]: UDPv4 link local: [undef]
Feb 14 12:57:11 openvpn[810]: UDPv4 link remote: [AF_INET]1xx.xx.247.13:1194
Feb 14 12:57:11 openvpn[810]: TLS_ERROR: BIO read tls_read_plaintext error: error:140830B5:SSL routines:ssl3_client_hello:no ciphers available
Feb 14 12:57:11 openvpn[810]: TLS Error: TLS object -> incoming plaintext read error
Feb 14 12:57:11 openvpn[810]: TLS Error: TLS handshake failed
Feb 14 12:57:11 openvpn[810]: SIGUSR1[soft,tls-error] received, process restarting
Feb 14 12:57:11 openvpn[810]: Restart pause, 2 second(s)
```

**Comment by angristan** [comment]
Try to remove the same line in your `/etc/openvpn/server.conf`

**Comment by NicolasLoew** [comment]
Nah still won't work

**Comment by NicolasLoew** [comment]
@Angristan Sup :P Just reporting back that I flushed my OS (it was a clean install aside from your script before though) and tried Nyrs script and it worked with the router out of the box. Just uploaded the ovpn file and done. Any idea why it doesn't with your version? I like yours because it leaves more possibilites to specify things :/

**Comment by angristan** [comment]
Hey @NicolasLoew I actually wanted you to try Nyr's, but forgot to send my message. :)

I think the other issue you were having was with tls-auth.

If you happen to try again, remove the `<tls-auth>` block in your client file and server conf

**Comment by NicolasLoew** [comment]
Aight! Thanks mate :)

**Comment by cezar97** [comment]
@NicolasLoew In fact, as Angristan's script is focused more on security, there are options that work only on recent versions. Nyr's one is based on defaults, that's why is compatible with a lot of machines.
According to https://community.openvpn.net/openvpn/wiki/Hardening, you have a openvpn version < v2.3.3, the first that introduced support for `--tls-version-min`. This means that it's very outdated and you can use only TLS 1.0. This means that `--tls-cipher` can use only TLSv1.0 cipher-suites, too. It seems that there aren't any issues with `--tls-auth`.
Can you paste the output of `openvpn --version`, just for reference? Also, someone reported that on Asus Routers, Nyr's script doesn't work anymore. Can you confirm that?

**Comment by cezar97** [comment]
Oh, there's also on the README:
https://github.com/Angristan/OpenVPN-install#tls-version at "TLS 1.2 is only supported since OpenVPN 2.3.3"...

**Comment by georgesotiriadis** [comment]
hello yes it does not i got the same router any suggestions so i can use my config?

**Comment by cezar97** [comment]
I wouldn't use TLS 1.0... So maybe compiling the latest version? Or using a low-cost raspberry-pi as a vpn server. This if Asus isn't going to update it soon.

**Comment by georgesotiriadis** [comment]
well my router its ac52u b1 i just bought im really dissapointed even without vpn i pluged in my 4g dongle and got 2 mbps what router you suggest me with openvpn i can change it from the store thats  why i need your suggestion if you could help me thanks

**Comment by cezar97** [comment]
Personally I don't recommend routers builtin vpn because with your own VPS or raspberry you can always get the latest version of openvpn and you don't load routers too much with encryption / decryption stuff, routers cpu don't really have that in mind. Even with the raspberry pi 3 b+ I don't think you're going to have a VPN server at 100MB/s because of this.
About the router choice... I think Asus routers are pretty good. So maybe you get only 2mbps beacuse of the current configuration. Some basic stuff is:
Disable things you dont need like NAS, DDNS etc.
Try to use 5Ghz instead of 2.4Ghz if you can ( clients support ).
Set channel bandwidth to 40Mhz or greater and choose a channel where there aren't many others ( auto should do it ).
Apart from that, assuming that with Ethernet you have greater speed, I don't really know what's the problem...

**Comment by georgesotiriadis** [comment]
Well i want to set up my 4g stick with opevpn will rasberry be the solution
? And can you provide me a link if possbile thanks

Στις Τρί, 26 Ιουν 2018, 1:06 π.μ. ο χρήστης cezar97 <
notifications@github.com> έγραψε:

**Comment by NicolasLoew** [comment]
I've achieved 20+ Mbit/s with ovpn on asus 51u so maybe check your setup again mate :)

**Comment by Papy-Serra** [comment]
Hi !

I actually got the same issue as you @NicolasLoew, I remove `tls-version-min` and `tls-auth`, but it doesn't work. I got the same error as you ` TLS_ERROR: BIO read tls_read_plaintext error:`

Here there are my config file, I got an ASUS Router RT-AC68U

Client Config:
```
client
proto udp
remote XXX.XXX.XXX.XXX 1194
dev tun
resolv-retry infinite
nobind
persist-key
persist-tun
remote-cert-tls server
auth SHA256
auth-nocache
cipher AES-128-CBC
tls-client
tls-cipher TLS-DHE-RSA-WITH-AES-128-GCM-SHA256
setenv opt block-outside-dns
verb 3
```

Server Config:
```
port 1194
proto udp
dev tun
user nobody
group nogroup
persist-key
persist-tun
keepalive 10 120
topology subnet
server X.X.X.X 255.255.255.0
ifconfig-pool-persist ipp.txt
push "dhcp-option DNS 8.8.8.8"
push "dhcp-option DNS 8.8.4.4"
push "redirect-gateway def1 bypass-dhcp" 
crl-verify crl.pem
ca ca.crt
cert server.crt
key server.key
dh dh.pem
auth SHA256
cipher AES-128-CBC
tls-server
tls-cipher TLS-DHE-RSA-WITH-AES-128-GCM-SHA256
status openvpn.log
verb 3
```

**Comment by angristan** [comment]
@Misteralex007 Can you send the full log?

**Comment by Papy-Serra** [comment]
Well I found a solution by installing OpenVPN from a older script (I saw this solution in a other issue) and it worked but the bandwith was very very slow. I mean with my computer with the OpenVPN client (Windows 10) I get full speed (80 Mbps i.e 10 mo/s) but with the router I barely get 2 Mbps.

I'm trying to configure a VPN Router/Client with a Raspberry Pi 2 Modele B with an Ethernet Adapter USB.

Thanks for your help @Angristan

**Comment by angristan** [comment]
Thanks for the feedback

**Comment by n0isecontrol** [comment]
Hey,
just wanted to stop by and tell how i solved that issue with your installer.
After removing the tls-cipher, tls-version-min options and changing the protocol to TCP it suddenly began to work with my ASUS RT-AC66U_B1. Not sure why UDP wouldnt work as it worked on my Debian machine meh.

**Comment by randshell** [comment]
@n0isecontrol yes, and that's completely insecure because it uses TLS1.0 :smiley:.

**Comment by coelho5br** [comment]
Try to update the firmware to Asuswrt-Merlin that supports new Open VPN versions
https://www.privateinternetaccess.com/blog/2018/08/hardware-acceleration-is-here-for-routers-using-openvpn/

**Comment by msaadbashir** [comment]
Hey I am still facing this issue on my ASUS RT-AC3200 Router (Firmware Version:3.0.0.4.382_51640). I used Cloudflare option and then the google option but both end of showing the same error.

It started with tls-version error. I removed that by opening the .ovpn file in text editor. And then it removed block-outside-dns and then it gave no closing quotation error.

What can i do to resolve this issue?

Following is the log 
`May 15 12:22:08 vpnclient1[21214]: Options error: Unrecognized option or missing parameter(s) in config.ovpn:36: ignore-unknown-option (2.3.2)
May 15 12:22:08 vpnclient1[21214]: Use --help for more information.
May 15 12:22:51 rc_service: httpd 300:notify_rc restart_vpncall
May 15 12:23:08 rc_service: httpd 300:notify_rc restart_vpncall
May 15 12:23:09 vpnclient1[21287]: Options error: Unrecognized option or missing parameter(s) in config.ovpn:36: block-outside-dns (2.3.2)
May 15 12:23:09 vpnclient1[21287]: Use --help for more information.
May 15 12:23:33 rc_service: httpd 300:notify_rc restart_vpncall
May 15 12:23:35 rc_service: httpd 300:notify_rc restart_vpncall
May 15 12:23:47 rc_service: httpd 300:notify_rc restart_vpncall
May 15 12:23:49 rc_service: httpd 300:notify_rc restart_vpncall
May 15 12:23:50 vpnclient1[21362]: Options error: No closing quotation (") in config.ovpn:36
May 15 12:23:50 vpnclient1[21362]: Use --help for more information.
